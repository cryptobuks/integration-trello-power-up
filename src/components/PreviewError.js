import React from 'react'

export default class PreviewError extends React.Component {
  render () {
    return (
      <p className='PreviewError basic-attachment-list-item'>
        <a
          href={this.props.attachment.url}
          target='_blank'
          rel='noopener noreferrer'
        >
          {this.props.attachment.url}
        </a>{' '}
        is not valid or password protected.{' '}
        <a
          target='_blank'
          rel='noopener noreferrer'
          href=''
          onClick={this.props.disconnect}
        >
          Disconnect prototype
        </a>{' '}
        or reach out to{' '}
        <a
          href='mailto:support@invisionapp.com'
          target='_blank'
          rel='noopener noreferrer'
        >
          support@invisionapp.com
        </a>{' '}
        for more information
      </p>
    )
  }
}
