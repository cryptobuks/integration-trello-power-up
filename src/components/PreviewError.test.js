import 'jsdom-global/register'
import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import PreviewError from './PreviewError.js'

global.window = {}

jest.mock('../util/trello.js')
jest.mock('../util/config.js')
jest.mock('../util/auth.js')
jest.mock('../util/cover.js')
jest.mock('../util/attachment.js')
jest.useFakeTimers()

const setup = () => {
  const ctx = {
    props: {
      attachment: {
        url: 'https://google.com'
      }
    }
  }
  ctx.wrapper = mount(<PreviewError {...ctx.props} />, {
    disableLifecycleMethods: false
  })

  ctx.component = ctx.wrapper.instance()
  expect(ctx.component).toBeDefined()
  expect(toJson(ctx.wrapper)).toMatchSnapshot()
  return ctx
}

describe('PreviewError', () => {
  it('should render', () => {
    setup()
  })
})
