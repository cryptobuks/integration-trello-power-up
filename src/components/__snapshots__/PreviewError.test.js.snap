// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewError should render 1`] = `
<PreviewError
  attachment={
    Object {
      "url": "https://google.com",
    }
  }
>
  <p
    className="PreviewError basic-attachment-list-item"
  >
    <a
      href="https://google.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      https://google.com
    </a>
     
    is not valid or password protected.
     
    <a
      href=""
      rel="noopener noreferrer"
      target="_blank"
    >
      Disconnect prototype
    </a>
     
    or reach out to
     
    <a
      href="mailto:support@invisionapp.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      support@invisionapp.com
    </a>
     
    for more information
  </p>
</PreviewError>
`;
